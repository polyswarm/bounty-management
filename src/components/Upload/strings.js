const strings = {
  selectFile: 'Select Files to Upload!'
}
export default strings;
