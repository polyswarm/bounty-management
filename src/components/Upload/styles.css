.hidden {
    width: 0.1px;
    height: 0.1px;
    overflow: hidden;
    opacity: 0;
    position: absolute;
    z-index: -1;
}

.hidden + label {
  font-family: 'fabrick', sans-serif;
  display: inline-block;
  height: 40px;
  margin-top: 20px;
  padding: 0 40px;
  line-height: 40px;
  font-size: 16px;
  background: rgb(46, 255, 109);
  color: #8500ff;
  border: 0;
  border-radius: 20px;
  text-shadow: none;
  text-transform: uppercase;
  text-decoration: none;
  letter-spacing: 0.15em;
  transition: background-color 0.25s ease;
  cursor: pointer;
}

.hidden + label:hover {
  background: #fafafa;
}

.hidden + label:active,
.hidden + label.selected {
  background: #e0e0e0;
}

.hidden:focus + label {
  background: #fafafa;
  outline: 1px solid rgb(46, 255, 109);
  outline: -webkit-focus-ring-color auto 5px;
}
