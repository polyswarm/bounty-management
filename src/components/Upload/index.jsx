import React, {Component} from 'react';
import './styles.css';
import strings from './strings';

class Upload extends Component {
  constructor(props) {
    super(props);
    this.state = {
      file: null,
    };
    this.onFileChanged = this.onFileChanged.bind(this);
  }

  render() {
    const {file} = this.state;
    return(
      <form>
        <input id='file' className='hidden' type='file' onChange={this.onFileChanged} />
        {file && (
          <React.Fragment>
            <label htmlFor='file' className='selected'>{file.split('\\').pop()}</label>
            <button>{strings.send}</button>
          </React.Fragment>
        )}
        {!file && (
          <label htmlFor='file'>{strings.selectFile}</label>
        )}
      </form>
    );
  }

  onFileChanged(event) {
    if (event.target.value) {
        this.setState({file: event.target.value})
    }
  }
}
export default Upload;
