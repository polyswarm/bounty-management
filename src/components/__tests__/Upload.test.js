import React from 'react';
import {render} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import Upload from '../App';

it('renders without crashing', () => {
  const wrapper = render(<Upload />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});
