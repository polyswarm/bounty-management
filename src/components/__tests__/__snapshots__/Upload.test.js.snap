// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="BountyManager hex-background"
>
  <p
    class="BountyManager-Intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <div
    class="Upload"
  >
    <div
      class="Drop-Target"
    >
      <p>
        Drag and drop your files here, or use the button below.
      </p>
    </div>
    <form>
      <input
        class="hidden"
        id="file"
        type="file"
      >
        
      </input>
      <label
        for="file"
      >
        Select Files to Upload!
      </label>
    </form>
  </div>
</div>
`;
